flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/TestBaseUtils.java

```diff
@@ -102,6 +116,24 @@ private static void verifyJvmOptions() {
 				+ "m", heap > MINIMUM_HEAP_SIZE_MB - 50);
 	}
 
+	public static void waitForAllTaskRunning(MiniClusterResource miniCluster, JobID jobID, Deadline deadline) throws Exception {
+		try (final RestClusterClient<?> clusterClient = new RestClusterClient<Object>(
+				miniCluster.getClientConfiguration(),
+				StandaloneClusterId.getInstance())) {
+			JobMessageParameters params = new JobMessageParameters();
+			params.jobPathParameter.resolve(jobID);
+			org.apache.flink.runtime.testutils.CommonTestUtils.waitUntilCondition(() -> {
+				final JobDetailsInfo jobDetailsInfo = clusterClient.sendRequest(
+					JobDetailsHeaders.getInstance(),
+					params,
+					EmptyRequestBody.getInstance()).get();
+				return jobDetailsInfo.getJobStatus() == JobStatus.RUNNING &&
+					jobDetailsInfo.getJobVerticesPerState().get(ExecutionState.RUNNING) ==
+						jobDetailsInfo.getJobVertexInfos().size();
+			}, deadline, 500);
+		}
+	}
+
```

Comment:
```suggestion
	public static void waitForAllTaskRunning(MiniCluster miniCluster, JobID jobID, Deadline deadline) throws Exception {
		org.apache.flink.runtime.testutils.CommonTestUtils.waitUntilCondition(() -> {
			AccessExecutionGraph ec = miniCluster.getExecutionGraph(jobID).get();
			return ec.getState() == JobStatus.RUNNING && ec.getAllVertices()
				.values()
				.stream()
				.allMatch(jobVertex ->
					Arrays.stream(jobVertex.getTaskVertices()).allMatch(task -> task.getExecutionState() == ExecutionState.RUNNING)
				);
		}, deadline);
	}
```

Reply:
Not sure about the code formatting, and I haven't tested it much. But I guess you get the idea.

Reply:
Hm Till actually proposed to use rest client, but I can't really see why your version wouldn't work.

Reply:
With your version, I can actually move it into `flink-runtime` `CommonTestUtils`.